### PR TITLE
initial backend implementation

### PR DESCRIPTION
**Pull request by arthosofteq**

### CODE REVIEW

File: redisinsight/ui/src/components/monitor-config/MonitorConfig.tsx [hunk 1]
```diff
@@ -59,7 +60,8 @@ const MonitorConfig = ({ retryDelay = 10000 } : IProps) => {
     let payloads: IMonitorDataPayload[] = []
 
     const handleMonitorEvents = () => {
-      newSocket.on(MonitorEvent.MonitorData, (payload:IOnDatePayload[]) => {
+      dispatch(setLoadingPause(false))
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/components/monitor-config/MonitorConfig.tsx:63]: Could you rename to "setMonitorLoadingPause"?

File: redisinsight/ui/src/components/monitor/Monitor/Monitor.tsx [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
   EuiTextColor,
   EuiToolTip,
 } from '@elastic/eui'
-import { AutoSizer } from 'react-virtualized'
+import AutoSizer from 'react-virtualized-auto-sizer'
 
 import { IMonitorDataPayload } from 'uiSrc/slices/interfaces'
 import { ReactComponent as BanIcon } from 'uiSrc/assets/img/monitor/ban.svg'
 
+import MonitorLog from '../MonitorLog/MonitorLog'
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/components/monitor/Monitor/Monitor.tsx:17]: Could you add index.ts? to escape duplication 'MonitorLog' in the path

File: redisinsight/ui/src/components/monitor/Monitor/Monitor.tsx [hunk 3]
```diff
@@ -67,20 +74,36 @@ const Monitor = (props: Props) => {
...
           </EuiFlexItem>
         </EuiFlexGroup>
       </div>
+      <div className={styles.saveLogContainer}>
+        <EuiToolTip
+          title="Allows you to download the generated log file after pausing the Profiler."
+          content="Profiler log is saved to a file on your local machine with no size limitation.
+          The temporary log file will be automatically rewritten when the Profiler is reset"
+        >
+          <>
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/components/monitor/Monitor/Monitor.tsx:90]: Please remove unnecessary empty Fragment

File: redisinsight/ui/src/components/monitor/MonitorLog/MonitorLog.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                  <EuiButton
+                    size="s"
+                    color="secondary"
+                    href={linkToDownload}
+                    iconType="download"
+                    className={styles.btn}
+                    data-testid="download-log-btn"
+                    {...downloadBtnProps}
+                  >
+                    {width > 360 && ' Download '}
```
Comment (zalenskiSofteq) [inline redisinsight/ui/src/components/monitor/MonitorLog/MonitorLog.tsx:86]: Could you move 360  and 460 to constants?

Comment (romansergeenkosofteq) [reply redisinsight/ui/src/components/monitor/MonitorLog/MonitorLog.tsx:86]: done

File: redisinsight/ui/src/components/monitor/Monitor/Monitor.tsx (hunk 3 above)
Comment (romansergeenkosofteq) [reply redisinsight/ui/src/components/monitor/Monitor/Monitor.tsx]: done

File: redisinsight/ui/src/components/monitor/Monitor/Monitor.tsx (hunk 2 above)
Comment (romansergeenkosofteq) [reply redisinsight/ui/src/components/monitor/Monitor/Monitor.tsx]: done

File: redisinsight/ui/src/components/monitor-config/MonitorConfig.tsx (hunk 1 above)
Comment (romansergeenkosofteq) [reply redisinsight/ui/src/components/monitor-config/MonitorConfig.tsx]: renamed

**Review by zalenskiSofteq — APPROVED** [review]

File: redisinsight/api/src/modules/profiler/models/log-file.ts [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  getReadStream(): ReadStream {
+    const stream = fs.createReadStream(this.filePath);
+    stream.once('end', () => {
+      stream.destroy();
+      try {
+        this.analyticsEvents.get(TelemetryEvents.ProfilerLogDownloaded)(this.instanceId, this.getFileSize());
+      } catch (e) {
+        // ignore analytics errors
+      }
+      // logFile.destroy();
```
Comment (AmirAllayarovSofteq) [inline redisinsight/api/src/modules/profiler/models/log-file.ts:65]: Can we remove this commented code?

**Review by AmirAllayarovSofteq — APPROVED** [review]

File: redisinsight/api/src/modules/profiler/models/log-file.ts (hunk 5 above)
Comment (arthosofteq) [reply redisinsight/api/src/modules/profiler/models/log-file.ts:65]: Nope :)

**Pull request merged**